### PR TITLE
Fix reboot of PVS-proxied VM

### PR DESCRIPTION
**Pull request by robhoes**

### CODE REVIEW

File: ocaml/xapi/monitor_pvs_proxy.ml [hunk 1]
```diff
@@ -85,7 +85,8 @@ let update () =
              let proxies = List.filter_map (fun vif -> Pvs_proxy_control.find_proxy_for_vif ~__context ~vif) vifs in
              let value = pvs_proxy_status_of_int status in
              List.iter (fun self ->
-                 Db.PVS_proxy.set_status ~__context ~self ~value
+                 if Db.PVS_proxy.get_currently_attached ~__context ~self then
+                   Db.PVS_proxy.set_status ~__context ~self ~value
```
Comment (mseri) [inline ocaml/xapi/monitor_pvs_proxy.ml]: What status appear when `currently_attached` is false?

File: ocaml/xapi/pvs_proxy_control.ml [hunk 2]
```diff
@@ -271,8 +271,6 @@ let stop_proxy ~__context vif proxy =
     error "Unable to disable PVS proxy for VIF %s: %s." (Ref.string_of vif) reason
 
 let clear_proxy_state ~__context vif proxy =
-    let site = Db.PVS_proxy.get_site ~__context ~self:proxy in
-    State.remove_proxy ~__context site vif;
     Db.PVS_proxy.set_currently_attached ~__context ~self:proxy ~value:false;
```
Comment (mseri) [inline ocaml/xapi/pvs_proxy_control.ml]: Does this mean that the ``stopped` status set below is not updated on the proxy status?

File: ocaml/xapi/monitor_pvs_proxy.ml (hunk 1 above)
Comment (robhoes) [inline ocaml/xapi/monitor_pvs_proxy.ml]: The `status` field is set to `stopped` when `currently_attached` is made `false` in `Pvs_proxy_control. clear_proxy_state`. The above conditional ensures that it does not get set to something else due to some race condition.

Comment (mseri) [inline ocaml/xapi/monitor_pvs_proxy.ml]: :+1:

**Pull request merged**